src/System.Private.CoreLib/shared/System/Number.BigInteger.cs

```diff
@@ -0,0 +1,749 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.InteropServices;
+using Internal.Runtime.CompilerServices;
+
+namespace System
+{
+    internal static partial class Number
+    {
+        [StructLayout(LayoutKind.Sequential, Pack = 1)]
+        internal unsafe ref struct BigInteger
+        {
+            private const int MaxBlockCount = 35;
+
+            private static readonly uint[] Pow10UInt32Table = new uint[]
+            {
+                1,          // 10^0
+                10,         // 10^1
+                100,        // 10^2
+                1000,       // 10^3
+                10000,      // 10^4
```

Comment:
Does CSC precalculate values like `10^4`, as it does for fields/locals?  I can't tell from the IL, but I guess so. Assuming so, you can just write `10^4`

Reply:
There isn't a way to express exponents for integer literals. `^` is the `xor` operator, so `10^4` would be `10 xor 4`.

Reply:
I verified that it does.

Reply:
Disregard my brainstorm above.